### PR TITLE
chore: add --inspect-file-watcher for watcherService

### PR DESCRIPTION
**Pull request by deepak1556**
Refs https://github.com/microsoft/vscode/issues/118613

### CODE REVIEW

File: src/vs/platform/files/node/diskFileSystemProvider.ts [hunk 1]
```diff
@@ -617,6 +621,7 @@ export class DiskFileSystemProvider extends Disposable implements
 						this.logService[msg.type](msg.message);
 					},
 					this.logService.getLevel() === LogLevel.Trace,
+					this.envArgs ? parseWatchServicePort(this.envArgs) : undefined,
```
Comment (deepak1556) [inline src/vs/platform/files/node/diskFileSystemProvider.ts:624]: In my local testing `this.envArgs` is `undefined` , @bpasero any idea what I missed in the initialization ?

File: src/vs/platform/files/node/diskFileSystemProvider.ts [hunk 2]
```diff
@@ -48,6 +51,7 @@ export class DiskFileSystemProvider extends Disposable implements
 
 	constructor(
 		private readonly logService: ILogService,
+		private readonly envArgs?: NativeParsedArgs,
```
Comment (bpasero) [inline src/vs/platform/files/node/diskFileSystemProvider.ts:54]: I would have rather expected to depend on `INativeEnvironmentService` here.

File: src/vs/platform/files/electron-browser/diskFileSystemProvider.ts [hunk 3]
```diff
@@ -18,7 +18,7 @@ export class DiskFileSystemProvider extends NodeDiskFileSystemProvider {
 		private readonly nativeHostService: INativeHostService,
 		options?: IDiskFileSystemProviderOptions
 	) {
-		super(logService, options);
+		super(logService, undefined, options);
```
Comment (bpasero) [inline src/vs/platform/files/electron-browser/diskFileSystemProvider.ts:21]: Shouldn't this pass the args to the super type? This is in the end the file system provider used for desktop VSCode?

File: src/vs/platform/files/node/diskFileSystemProvider.ts (hunk 1 above)
Comment (bpasero) [reply src/vs/platform/files/node/diskFileSystemProvider.ts:624]: Yes because you pass `undefined` in the disk file system provider we use on desktop.

**Review by bpasero — CHANGES_REQUESTED** [review]
If we change `DiskFileSystemProvider` to accept the native environment service you might also need to update our remote server where the provider is used.

Can I learn more about the rationale of this change? Why do we need to debug the watcher exactly?

**Comment by deepak1556** [comment]
> Can I learn more about the rationale of this change? Why do we need to debug the watcher exactly?

When we get report of high cpu usage with file watcher, I don't have a way to get CPU profile.

File: src/vs/platform/files/electron-browser/diskFileSystemProvider.ts (hunk 3 above)
Comment (deepak1556) [reply src/vs/platform/files/electron-browser/diskFileSystemProvider.ts:21]: Ah okay, I am unfamiliar with this code, will fix that.

**Comment by bpasero** [comment]
@deepak1556 well at least for when `chokidar` is used the explanation is quite simple: `chokidar` does a full folder scan over all files and folders right on startup. The CPU should eventually settle though and not stay high. If it stays high, this would hint at slow polling being used which I think @aeschli is doing in remote, maybe he can clarify.

**Comment by deepak1556** [comment]
Thanks for the context!

For the cases with constant high cpu usage and is not remote scenario, what is usual flow of debugging ? I was mainly targeting this PR for those unknown scenarios.

**Comment by bpasero** [comment]
@deepak1556 yeah for those cases we have no way of debugging, so we can use this change to create perf profiles I guess?

**Comment by deepak1556** [comment]
Yep thats correct

**Pull request closed without merging**